src/main/java/be/seeseemelk/mockbukkit/entity/EntityMock.java

```diff
@@ -28,552 +16,488 @@
+                return true;
+        }
+        return false;
+    }
+
+    @Override
+    public boolean hasPermission(String name) {
+        for (PermissionAttachment attachment : permissionAttachments) {
+            Map<String, Boolean> permissions = attachment.getPermissions();
+            if (permissions.containsKey(name) && permissions.get(name))
+                return true;
+        }
+        return false;
+    }
+
+    @Override
+    public boolean hasPermission(Permission perm) {
+        return hasPermission(perm.getName());
+    }
+
+    @Override
+    public PermissionAttachment addAttachment(Plugin plugin, String name, boolean value) {
+        PermissionAttachment attachment = addAttachment(plugin);
+        attachment.setPermission(name, value);
+        return attachment;
+    }
+
+    @Override
+    public PermissionAttachment addAttachment(Plugin plugin) {
+        PermissionAttachment attachment = new PermissionAttachment(plugin, this);
+        permissionAttachments.add(attachment);
+        return attachment;
+    }
+
+    @Override
+    public PermissionAttachment addAttachment(Plugin plugin, String name, boolean value, int ticks) {
+        // TODO Auto-generated constructor stub
+        throw new UnimplementedOperationException();
+    }
+
+    @Override
+    public PermissionAttachment addAttachment(Plugin plugin, int ticks) {
+        // TODO Auto-generated constructor stub
+        throw new UnimplementedOperationException();
+    }
+
+    @Override
+    public void removeAttachment(PermissionAttachment attachment) {
+        // TODO Auto-generated constructor stub
+        throw new UnimplementedOperationException();
+
+    }
+
+    @Override
+    public void recalculatePermissions() {
+
+    }
+
+    @Override
+    public Set<PermissionAttachmentInfo> getEffectivePermissions() {
+        // TODO Auto-generated constructor stub
+        throw new UnimplementedOperationException();
+    }
+
+    @Override
+    public String getCustomName() {
+        return customName;
+    }
+
+    @Override
+    public void setCustomName(String name) {
+        customName = name;
+    }
+
+    @Override
+    public Vector getVelocity() {
+        // TODO Auto-generated constructor stub
+        throw new UnimplementedOperationException();
+    }
+
+    @Override
+    public void setVelocity(Vector velocity) {
+        // TODO Auto-generated constructor stub
+        throw new UnimplementedOperationException();
+    }
+
+    @Override
+    public boolean isOnGround() {
+        // TODO Auto-generated constructor stub
+        throw new UnimplementedOperationException();
+    }
+
+    @Override
+    public List<Entity> getNearbyEntities(double x, double y, double z) {
+        // TODO Auto-generated constructor stub
+        throw new UnimplementedOperationException();
+    }
+
+    @Override
+    public int getEntityId() {
+        return new ArrayList<>(MockBukkit.getMock().getEntities()).indexOf(this);
```

Comment:
hmm, what if i stored the index when the class is constructed and then i just return that stored index

Reply:
This would not fix the problem with removing entities from list:
- we add two entities to a list and they get id 1 and 2 (2 entities in a list, the next one would get id = 3)
- we remove entity number 1 (1 entity in a list with id = 2, the next one would get id = 2)
- we add new entity, but this entity gets index number 2 and we have two entities with id = 2

Better than storing index would be storing the value that would be never the same, like this code would give us:
```java
private static int nextId = 1; //static value has only one instance for all objects in a class

private static int getNextId(){
    return nextId++;  //post-incrementation
}
```